message_test.go

```diff
@@ -705,6 +709,7 @@ func getBoundaries(t *testing.T, count int, m string) []string {
 var boundaryRegExp = regexp.MustCompile("boundary=(\\w+)")
 
 func mockCopyFile(name string) (string, FileSetting) {
+	os.Create(filepath.Base(name))
```

Comment:
Correct me if I'm wrong, but won't this also create a bunch of files in the working directory? We need a teardown function to remove them when the test suite finishes. Also, this ends up being a step away from an *actual* `copyFile` and not a mock.

I need to think about this more. I'll come back after I've reviewed some of the other pull requests.